tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    this.on( 'testAction', () => {
+        assert.ok( 'The test action was called' );
+    });
+
+    this.$( '.sl-calendar .active' ).click();
+});
+
+test( 'Action passes through expected objects in content array', function( assert ) {
+
+    assert.expect( 8 );
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', multipleTestContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    this.on( 'testAction', ( dateContent ) => {
+
+        // This group of asserts verifies the multiple events on Sep. 17th
+        if ( 'Event 1 Today!' === dateContent[ 0 ].label ) {
+            assert.equal(
+                dateContent[ 0 ].date.toString(),
+                'Sat Sep 17 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 0 ].label,
+                'Event 1 Today!',
+                'The label property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 1 ].date.toString(),
+                'Sat Sep 17 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 1 ].label,
+                'Event 2 Today!',
+                'The label property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 2 ].date.toString(),
+                'Sat Sep 17 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 2 ].label,
+                'Event 3 Today!',
+                'The label property was passed through'
+            );
+        // This group of asserts verifies the event on Sep. 20th
+        } else {
+            assert.equal(
+                dateContent[ 0 ].date.toString(),
+                'Tue Sep 20 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 0 ].label,
+                'Event 1 Another Day!',
+                'The label property was passed through'
+            );
+        }
+    });
+
+    this.$( '.sl-calendar .active' ).click();
+});
+
+test( 'Setting viewMode modifies the view correctly', function( assert ) {
```

Comment:
Would prefer this to either be broken into two tests or this one be expanded to cover more scenarios.  Essentially, it is possible for this component to render by only calling `{{sl-calender}}`, defaulting to the `days` view, which should be asserted.  Also asserted should be the setting of `viewMode` to `months` and 'years'.  All of these are without any date values passed to the component.

*Then* we also want to assert that in addition to the `viewMode` specified, it correctly highlights the day, month or year information that is provided to `content`.